src/AbstractDoctrineAnnotationFixer.php

```diff
@@ -0,0 +1,234 @@
+        foreach ($phpTokens->findGivenKind(T_DOC_COMMENT) as $index => $docCommentToken) {
+            if (!$this->nextElementAcceptsDoctrineAnnotations($phpTokens, $index)) {
+                continue;
+            }
+
+            $tokens = $this->getDoctrineAnnotationTokens($docCommentToken);
+            $this->fixAnnotations($tokens);
+            $docCommentToken->setContent($tokens->getCode());
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(PhpTokens $tokens)
+    {
+        return $tokens->isTokenKindFound(T_DOC_COMMENT);
+    }
+
+    /**
+     * Fixes Doctrine annotations from the given PHPDoc style comment.
+     *
+     * @param DoctrineAnnotationTokens $doctrineAnnotationTokens
+     */
+    abstract protected function fixAnnotations(DoctrineAnnotationTokens $doctrineAnnotationTokens);
+
+    /**
+     * @return string[]
+     */
+    protected function getDefaultIgnoredTags()
+    {
+        return array_merge(Tag::getTags(), array(
+            // PHPUnit
+            'after',
+            'afterClass',
+            'backupGlobals',
+            'backupStaticAttributes',
+            'before',
+            'beforeClass',
+            'codeCoverageIgnore',
+            'codeCoverageIgnoreStart',
+            'codeCoverageIgnoreEnd',
+            'covers',
+            'coversDefaultClass',
+            'coversNothing',
+            'dataProvider',
+            'depends',
+            'expectedException',
+            'expectedExceptionCode',
+            'expectedExceptionMessage',
+            'expectedExceptionMessageRegExp',
+            'group',
+            'large',
+            'medium',
+            'preserveGlobalState',
+            'requires',
+            'runTestsInSeparateProcesses',
+            'runInSeparateProcess',
+            'small',
+            'test',
+            'testdox',
+            'ticket',
+
+            // other
+            'TODO',
+            'todo',
+            'FIXME',
+            'fixme',
+        ));
+    }
+
+    /**
+     * @param PhpTokens $tokens
+     * @param int       $index
+     *
+     * @return bool
+     */
+    private function nextElementAcceptsDoctrineAnnotations(PhpTokens $tokens, $index)
+    {
+        do {
+            $index = $tokens->getNextMeaningfulToken($index);
+        } while ($tokens[$index]->isGivenKind(array(T_ABSTRACT, T_FINAL)));
+
+        if ($tokens[$index]->isClassy()) {
+            return true;
+        }
+
+        while ($tokens[$index]->isGivenKind(array(T_PUBLIC, T_PROTECTED, T_PRIVATE, T_FINAL, T_ABSTRACT))) {
+            $index = $tokens->getNextMeaningfulToken($index);
+        }
+
+        $analyzer = new TokensAnalyzer($tokens);
+
+        return array_key_exists($index, $analyzer->getClassyElements());
+    }
+
+    /**
+     * @param PhpToken $input
+     *
+     * @return DoctrineAnnotationTokens
+     */
+    private function getDoctrineAnnotationTokens(PhpToken $input)
```

Comment:
I would move it outside of this class (eg into `Annotation\Tokens::createFrom...`)